crates/ty_ide/src/symbols.rs

```diff
@@ -516,81 +878,164 @@ impl SourceOrderVisitor<'_> for SymbolVisitor {
                     return;
                 }
 
+                // Mark that we're entering a class scope
+                let was_in_class = self.in_class;
+                self.in_class = true;
```

Comment:
Depending on your definition of `in_class`, you'll want to unset `in_class` when entering a new function and unset `in_function` when entering a class. You might also want to unset `in_class` and `in_function` when entering a lambda expression

Reply:
I don't think we need this level of precision here. I added some comments to clarify:

```rust
    /// Track if we're currently inside a function at any point.
    ///
    /// This is true even when we're inside a class definition
    /// that is inside a class.
    in_function: bool,
    /// Track if we're currently inside a class at any point.
    ///
    /// This is true even when we're inside a function definition
    /// that is inside a class.
    in_class: bool,
```